packages/dds/matrix/src/handlecache.ts

```diff
@@ -100,14 +106,18 @@ export class HandleCache implements IVectorConsumer<Handle> {
 		if (_position < this.start) {
 			this.handles = this.getHandles(_position, this.start).concat(this.handles);
 			this.start = _position;
-			return this.handles[0];
+			// TODO why are we non null asserting here?
```

Comment:
`getHandles` can return an empty array, so we might want to add checks that handles[0] is not undefined instead of asserting. Not for this pr, though, just a suggestion for later.

Reply:
(applies to all the below assertions after a `getHandles` call)